File: Ejercicios/Facil/06 - FACTORIAL RECURSIVO/06 - FACTORIAL RECURSIVO.py
# Escribe una función que calcule y retorne el factorial
# de un número dado de forma recursiva.

# (n-1)
#def multiplicacion(n):
#
#    for item in range(1, n):
#        item = 1
#        n -= item
#        result = n
#        print(f"Numero: {result}")
#
#    return result
#
# n(n-1)!
def factorial(n):
    if(n==0):
        return 1
    else:
        result = n * factorial(n-1)
        print(result)
        return result

#    for item in range(n):
#        n =- item
#        result = n * multiplicacion(n)
#        print(result)
#    return result

# Solicitar valor al usuario
n = int(input("Ingresa un numero, para factorizar:  "))
print(f"El factorial de: {n} es: {factorial(n)}")
#factorial(5)

File: Ejercicios/Facil/06 - FACTORIAL RECURSIVO/06 - FACTORIAL RECURSIVO.js
/*
 * Escribe una función que calcule y retorne el factorial de un número dado
 * de forma recursiva.
 */

let res;
const factorial = (num) => {
  if (num === 0) {
    return 1;
  } else {
    res = num * factorial(num - 1);
  
    return res;
  }
  
};

console.log(factorial(9));
